src/System.IO.Pipes.AccessControl/tests/NamedPipeTests/NamedPipeTest.AclExtensions.cs

```diff
@@ -40,11 +40,27 @@ public void GetAccessControl_NamedPipe_BeforeWaitingToConnect()
             Assert.Throws<InvalidOperationException>(() => client.SetAccessControl(new PipeSecurity()));
         }
 
+        [Fact]
+        [SkipOnTargetFramework(TargetFrameworkMonikers.NetFramework, "On Desktop we throw an exception based on the internal PipeStream state. In .NET Core, we do not have access to that member since Pipes.AccessControl is its own assembly.")]
+        public void SetAccessControl_NamedPipeClientStream_Broken()
+        {
+            NamedPipeClientStream client = GetAccessControl_NamedPipeClientStream_Broken();
+            Assert.Throws<InvalidOperationException>(() => client.SetAccessControl(new PipeSecurity()));
+        }
+
+        [Fact]
+        [SkipOnTargetFramework(~TargetFrameworkMonikers.NetFramework, "On Desktop we throw an exception based on the internal PipeStream state. In .NET Core, we do not have access to that member since Pipes.AccessControl is its own assembly.")]
```

Comment:
As an aside from this PR, we should consider adding to our xunit extensions an InlineDataTargetFramework attribute, so that we don't have to duplicate tests like this.  In that case it would become something like:
```C#
[Theory]
[InlineDataTargetFramework(TargetFrameworkMonikers.NetFramework, typeof(InvalidOperationException))]
[InlineDataTargetFramework(TargetFrameworkMonikers.~NetFramework, typeof(IOException)]
public void SetAccessControl_NamedPipeClientStream_Broken_Desktop(Type expectedException)
{
    ...
    Assert.Throws(expectedException, () => ...);
}
```
Alternatively and easier, we could just add an AssertThrows helper that takes the expected exception on core and the expected exception on desktop, and verifies the right one based on the platform.  We have a helper like that used for parameter names.